test/fastify-plugins/maps.js

```diff
@@ -223,6 +224,79 @@ test('/style.json resolves style.json of fallback map when custom and online are
   )
 })
 
+test('custom map info endpoint not available when custom map path not provided', async (t) => {
+  const server = setup(t)
+
+  server.register(MapServerPlugin, {
+    defaultOnlineStyleUrl: DEFAULT_ONLINE_STYLE_URL,
+    fallbackMapPath: DEFAULT_FALLBACK_MAP_FILE_PATH,
+  })
+
+  const address = await server.listen()
+
+  mockAgent.enableNetConnect(new URL(address).host)
+
+  const response = await server.inject({
+    method: 'GET',
+    url: `/${CUSTOM_MAP_PREFIX}/info`,
+  })
+
+  assert.equal(response.statusCode, 404)
+})
+
+test('custom map info endpoint returns expected error when custom map is invalid', async (t) => {
+  const server = setup(t)
+
+  const nonExistentFile =
+    path.parse(SAMPLE_SMP_FIXTURE_PATH).dir + '/does/not/exist.smp'
+
+  server.register(MapServerPlugin, {
+    customMapPath: nonExistentFile,
+    defaultOnlineStyleUrl: DEFAULT_ONLINE_STYLE_URL,
+    fallbackMapPath: DEFAULT_FALLBACK_MAP_FILE_PATH,
+  })
+
+  const address = await server.listen()
+
+  mockAgent.enableNetConnect(new URL(address).host)
+
+  const response = await server.inject({
+    method: 'GET',
+    url: `/${CUSTOM_MAP_PREFIX}/info`,
+  })
+
+  assert.equal(response.statusCode, 500, 'returns server error status code')
+})
+
+test('custom map info endpoint returns expected info when available', async (t) => {
+  const server = setup(t)
+
+  server.register(MapServerPlugin, {
+    customMapPath: SAMPLE_SMP_FIXTURE_PATH,
+    defaultOnlineStyleUrl: DEFAULT_ONLINE_STYLE_URL,
+    fallbackMapPath: DEFAULT_FALLBACK_MAP_FILE_PATH,
+  })
+
+  const address = await server.listen()
+
+  mockAgent.enableNetConnect(new URL(address).host)
+
+  const response = await server.inject({
+    method: 'GET',
+    url: `/${CUSTOM_MAP_PREFIX}/info`,
+  })
+
+  assert.equal(response.statusCode, 200)
+
+  const info = response.json()
+
+  assert.deepEqual(info, {
+    created: '2024-10-09T17:41:16.103Z',
+    size: 2815018,
+    name: 'MapLibre',
+  })
```

Comment:
any thoughts on the best approach for this assertion? seems like `created` field is different each time github actions run so it always seems to fail (maybe that's expected??)

Reply:
Stat the file in the test that you are passing to the server, and check the value from the server is the same as what you get from the file.